core/src/sumeragi/main_loop.rs

```diff
@@ -482,7 +474,7 @@ fn cache_transaction<F: FaultInjection>(state: &mut State, sumeragi: &SumeragiWi
 fn suggest_view_change<F: FaultInjection>(
     sumeragi: &SumeragiWithFault<F>,
     state: &State,
-    view_change_proof_chain: &mut Vec<Proof>,
+    view_change_proof_chain: &mut ProofChain,
```

Comment:
would it make sense to put `view_change_proof_chain` into `state`?

Reply:
We probably can move all `mut` variables present in `run` function into `State`, this change can help with reducing number of argument we pass into functions.
One other possible benefit would be that we can log entire state with every loop iteration for example.